src/main/java/it/gov/pagopa/reward/service/counters/UserInitiativeCountersUnlockMediatorServiceImpl.java

```diff
@@ -0,0 +1,42 @@
+package it.gov.pagopa.reward.service.counters;
+
+import it.gov.pagopa.reward.connector.repository.UserInitiativeCountersRepository;
+import it.gov.pagopa.reward.dto.trx.RewardTransactionDTO;
+import it.gov.pagopa.reward.model.counters.UserInitiativeCounters;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+import reactor.core.publisher.Mono;
+
+import java.util.List;
+
+import static it.gov.pagopa.reward.utils.RewardConstants.PAYMENT_STATE_AUTHORIZED;
+import static it.gov.pagopa.reward.utils.RewardConstants.PAYMENT_STATE_REWARDED;
+
+@Service
+@Slf4j
+public class UserInitiativeCountersUnlockMediatorServiceImpl implements UserInitiativeCountersUnlockMediatorService {
+
+    private static final List<String> ACCEPTED_STATUS = List.of(PAYMENT_STATE_AUTHORIZED, PAYMENT_STATE_REWARDED);
+    private final UserInitiativeCountersRepository userInitiativeCountersRepository;
+
+    public UserInitiativeCountersUnlockMediatorServiceImpl(UserInitiativeCountersRepository userInitiativeCountersRepository){
+        this.userInitiativeCountersRepository = userInitiativeCountersRepository;
+
+    }
+
+    @Override
+    public Mono<UserInitiativeCounters> execute(RewardTransactionDTO transactionDTO) {
+        return Mono.just(transactionDTO)
+                .filter(trx -> ACCEPTED_STATUS.contains(trx.getStatus()))
+                .flatMap(this::handlerUnlockType);
+    }
+
+    private Mono<UserInitiativeCounters> handlerUnlockType(RewardTransactionDTO trx) {
+        if(PAYMENT_STATE_AUTHORIZED.equals(trx.getStatus())
+            || PAYMENT_STATE_REWARDED.equals(trx.getStatus())) {
+            return userInitiativeCountersRepository.unlockPendingTrx(trx.getId());
+        }
+        //TODO handle expired event (CANCELLED status)
```

Comment:
Are you going to resolve the `TODO` in this PR?
If not i suggest to remove it and create a new jira task instead to accomplish the future changes. (add the reference to the codebase in the task)

Reply:
This TODO regards the handling of AUTHORIZATION_REQUESTED expiration, which will be introduced the next Sprint.
I suggested to design the solution in order to take into account this future work: as a result, this TODO is just put a reminder on the code in order to let to know where to start the implementation of the (we could put inside the comment the id of the future task if you prefer)

Reply:
yes please, link the code reference in the jira task regarding the feature. The TODO comment tend to be forgotten there :)